lib/job_base.cu

```diff
@@ -0,0 +1,71 @@
+#include <include/job_base.h>
+
+// constructor and initialization
+JobBase::JobBase(){
+    partition = 1 / size_of_process_time;
+}
+
+__device__ __host__ void JobBase::init(){
+    this->partition = 0;
```

Comment:
init function is used to initialize some kind of variables, such as `this->start_time`, `this->end_time`, which are changed on last iteration of genetic algorithm.
` this->partition` and `this->arriv_t `should not be assigned 0.